Add setting to cache the GPUMemory mappings for pipeline upload

samikhawaja opened the pull request:
During pipeline upload the GpuMemory is allocated from the pool. This
memory is mapped, pipeline is uploaded and then the memory is unmapped.
In applications with lots of pipelines being created during rendering,
this causes lots of map/unmap calls and slows down the pipeline upload.

Add a setting that allows the cache of the GpuMemory mapping, so that it
can be reused by another pipeline upload.

samikhawaja commented:
Fixes: https://github.com/GPUOpen-Drivers/pal/issues/72

amdvlk-admin commented:
Can one of the admins verify this patch?

JaxLinAMD commented:
ok to test

JakeMerdichAMD commented:
I'm in favor of this change, but am also concerned that this will grow unboundedly-- many of our systems in the wild still have limited BAR space and thus limited GPU memory that they can CPU-map at a time.

Either only doing this when we don't have this limitation (no invisible memory) or having some scheme to eventually unmap the memory so we don't have this issue would resolve this in my mind.

samikhawaja commented:
The mapped memory is unmapped when the internal memory manager is destroyed. But in general since the maximum pool size is large enough the later memory allocations will be serviced by already allocated pools.

Also if needed to enable this only when required, I have added a new pal setting that disables it by default.

RobMartin-AMD commented:
> The mapped memory is unmapped when the internal memory manager is destroyed. But in general since the maximum pool size is large enough the later memory allocations will be serviced by already allocated pools.
> 
> Also if needed to enable this only when required, I have added a new pal setting that disables it by default.

Some AAA game titles have very large numbers of pipelines (sometimes tens of thousands).  Even with large pools of memory, there could still be a large number of these pools active simultaneously.  Keeping these allocations mapped essentially forever (because the internal memory mgr is not destroyed until the parent `Device` is destroyed) can be detrimental on some platforms.  which this code base has to support.  

Having the setting is nice, but a better option would be to have a limit on the number of these large pools which can remain mapped.  If we maintained a list of open mappings, and released the least-recently-used mapping when we exceeded our maximum, we'd be able to have most of if not all of the benefits of this change, without any of the potential drawbacks.  If this maximum were controllable via a PAL setting, then we could set the maximum to zero in cases where we might need to disable this functionality.

samikhawaja commented:
> > The mapped memory is unmapped when the internal memory manager is destroyed. But in general since the maximum pool size is large enough the later memory allocations will be serviced by already allocated pools.
> > Also if needed to enable this only when required, I have added a new pal setting that disables it by default.
> 
> Some AAA game titles have very large numbers of pipelines (sometimes tens of thousands). Even with large pools of memory, there could still be a large number of these pools active simultaneously. Keeping these allocations mapped essentially forever (because the internal memory mgr is not destroyed until the parent `Device` is destroyed) can be detrimental on some platforms which this code base has to support.
> 
> Having the setting is nice, but a better option would be to have a limit on the number of these large pools which can remain mapped. If we maintained a list of open mappings, and released the least-recently-used mapping when we exceeded our maximum, we'd be able to have most of if not all of the benefits of this change, without any of the potential drawbacks. If this maximum were controllable via a PAL setting, then we could set the maximum to zero in cases where we might need to disable this functionality.

Makes sense. I have updated this to use a maxNumMappedPool setting to decide the maximum number of mapped pools. Pools are unmapped based on least recently used scheme.

amdvlk-admin commented:
### Test summary for commit 53cefe36af9ddea984d003db95a2fb856117b334
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 72afd4ae01e3f5c043dbd2dfac57833408ac3f6b</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 01f326c1e88cbf80e50a631ea0ce61a57152a7c0</li>
</ul></details>

<details><summary>CTS tests (Failed: 336/227984)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31596/56996 (55.4%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 155


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 25316/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31593/56996 (55.4%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 550


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 25319/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31967/56996 (56.1%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 155


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 24945/56996 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34848/56996 (61.1%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 155


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 22064/56996 (38.7%)</li>
</ul></details>


</ul></details>

amdvlk-admin commented:
### Test summary for commit fa620f7bf73f48ef75e53d88f0508652568ed566
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 72afd4ae01e3f5c043dbd2dfac57833408ac3f6b</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 01f326c1e88cbf80e50a631ea0ce61a57152a7c0</li>
</ul></details>

<details><summary>CTS tests (Failed: 336/227984)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31596/56996 (55.4%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 155


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 25316/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31593/56996 (55.4%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 550


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 25319/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31967/56996 (56.1%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 155


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 24945/56996 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34848/56996 (61.1%)</li>
<li><details><summary>Failed: 84/56996 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Script:
Allocations still remain, failed on index 155


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.instance
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_group
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.device_memory_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_uniform_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_small
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_storage_large
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_3d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_1d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_2d_arr
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube
Stack trace: Script:
Invalid allocation callback


FAILURE: dEQP-VK.api.object_management.single_alloc_callbacks.image_view_cube_arr
Stack trace: Script:
Invalid allocation callback

...
```
</p></details></li>

<li>Not Supported: 22064/56996 (38.7%)</li>
</ul></details>


</ul></details>

amdvlk-admin commented:
### Test summary for commit a19f205cc40c3c166324c0b243737632a93504ce
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 72afd4ae01e3f5c043dbd2dfac57833408ac3f6b</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 01f326c1e88cbf80e50a631ea0ce61a57152a7c0</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/227984)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31680/56996 (55.6%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 25316/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31677/56996 (55.6%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 25319/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32051/56996 (56.2%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 24945/56996 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34932/56996 (61.3%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 22064/56996 (38.7%)</li>
</ul></details>


</ul></details>

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +576,116 @@ uint32 InternalMemMgr::GetReferencesCount()
+                break;
+            }
+        }
+    }
```
If `WasBuddyAllocated()` returns false, the memory will never be unmapped.  There should be an `else` case which unmaps the `IGpuMemory` object directly.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +576,116 @@ uint32 InternalMemMgr::GetReferencesCount()
+Result InternalMemMgr::Unmap(
+        GpuMemory* pGpuMemory)
+{
+    Util::MutexAuto allocatorLock(&m_allocatorLock); // Ensure thread-safety using the lock
```
Calling `WasBuddyAllocated()` is fine to do outside the lock, so the lock should only be taken inside the scope of that check.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.h:
```diff
@@ -56,6 +56,9 @@ struct GpuMemoryPool
     uint64                          pagingFenceVal;         // Paging fence value
 
     Util::BuddyAllocator<Platform>* pBuddyAllocator;        // Buddy allocator used for the suballocation
+    bool                            mapped;                 // flag to check if this pool is already mapped
```
I don't think you need this flag.  If `pData` is `nullptr`, then the memory is not already mapped.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +576,116 @@ uint32 InternalMemMgr::GetReferencesCount()
+        GpuMemory* pGpuMemory,
+        void** ppData)
+{
+    Util::MutexAuto allocatorLock(&m_allocatorLock); // Ensure thread-safety using the lock
```
Calling `WasBuddyAllocated()` is fine to do outside the lock, so the lock should only be taken inside the scope of that check.

samikhawaja replied:
Done

RobMartin-AMD reviewed (commented):
This change doesn't conform to the PAL coding standards.  Please update it so that it does.  You can review the standards here:  https://github.com/GPUOpen-Drivers/pal/blob/dev/doc/process/palCodingStandards.md

samikhawaja commented:
I have also change the max logic from number of pools to the size of mapped pools.

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,123 @@ uint32 InternalMemMgr::GetReferencesCount()
+    return result;
+}
+
+Result InternalMemMgr::Unmap(
+        GpuMemory* pGpuMemory)
```
Nit: Please match the header-comment style we use throughout these *.cpp files at the top of each function body.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,123 @@ uint32 InternalMemMgr::GetReferencesCount()
+    return Result::Success;
+}
+
+void InternalMemMgr::CheckMappedPoolLimit()
```
Nit: Please match the header-comment style we use throughout these *.cpp files at the top of each function body.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,123 @@ uint32 InternalMemMgr::GetReferencesCount()
+{
+    if ((m_pDevice->Settings().maxMappedPoolsSize >= 0))
+    {
+        while (m_totalSizeMappedPools > m_pDevice->Settings().maxMappedPoolsSize
+                && m_unusedMappedPoolList.NumElements() > 0)
```
Nit:  Please add parentheses around sub-expressions.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,123 @@ uint32 InternalMemMgr::GetReferencesCount()
+            GpuMemoryPool *pPool = *it.Get();
+
+            PAL_ASSERT(pPool->pBuddyAllocator != nullptr);
+            if (pPool->pData != nullptr && pPool->pGpuMemory != nullptr)
```
Nit:  Please add parentheses around sub-expressions.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,123 @@ uint32 InternalMemMgr::GetReferencesCount()
+
+void InternalMemMgr::CheckMappedPoolLimit()
+{
+    if ((m_pDevice->Settings().maxMappedPoolsSize >= 0))
```
FYI:  It looks like you accidentally put two sets of parentheses around this conditional.

samikhawaja replied:
Done

RobMartin-AMD commented on src/core/settings_core.json:
```diff
@@ -1881,6 +1881,23 @@
+        "Performance"
+      ],
+      "Defaults": {
+        "Default": 0
```
Thanks for leaving the default at zero here!  This minimizes the risk of regressions being introduced and gives the opportunity for later tuning.

samikhawaja replied:
Yes, during my earlier testing I discovered that some the CTS tests were failing, but I have already update the change to fix that.

RobMartin-AMD reviewed (commented):
Mostly looks good, just a few minor formatting nits.  I realize these may seem arbitrary or unnecessary, but the rationale behind them is to enforce a consistent style throughout all of PAL.  Doing so helps maintain the readability of the codebase for all developers, internal _and_ external to AMD.   It is much appreciated.

Thanks for your patience on this.

samikhawaja commented on src/core/internalMemMgr.cpp:
```diff
@@ -138,6 +139,24 @@ InternalMemMgr::InternalMemMgr(
+    {
+        PAL_ASSERT((it.Get() != nullptr) && (it.Get()->pBuddyAllocator != nullptr));
+
+        if (it.Get()->pData != nullptr && it.Get()->pGpuMemory != nullptr)
```
Done

samikhawaja commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,123 @@ uint32 InternalMemMgr::GetReferencesCount()
     return static_cast<uint32>(m_references.NumElements());
 }
 
+Result InternalMemMgr::Map(
+        GpuMemory* pGpuMemory,
+        void** ppData)
```
Done

samikhawaja commented:
> Mostly looks good, just a few minor formatting nits. I realize these may seem arbitrary or unnecessary, but the rationale behind them is to enforce a consistent style throughout all of PAL. Doing so helps maintain the readability of the codebase for all developers, internal _and_ external to AMD. It is much appreciated.
> 
> Thanks for your patience on this.

Makes perfect sense, really appreciate the detailed review.

RobMartin-AMD reviewed (commented):
Changes look good to me.

@jinjianrong -- What additional testing (if any) is required before this can be marked as approved?

JaxLinAMD commented:
I 'll re-trigger the test ( retest this please)
@JakeMerdichAMD @RobMartin-AMD please approve it if there is no problem

amdvlk-admin commented:
### Test summary for commit 12c9330bccb8c3900aa002694127516f0eeecd64
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 7569f1e64a840f4918ea6549ad3b9630758c6de9</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 4f968af13313fd60f4a661943745d252ad14f2af</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225188)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31329/56297 (55.6%)</li>
<li>Failed: 0/56297 (0.0%)</li>
<li>Not Supported: 24968/56297 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31329/56297 (55.6%)</li>
<li>Failed: 0/56297 (0.0%)</li>
<li>Not Supported: 24968/56297 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31712/56297 (56.3%)</li>
<li>Failed: 0/56297 (0.0%)</li>
<li>Not Supported: 24585/56297 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34555/56297 (61.4%)</li>
<li>Failed: 0/56297 (0.0%)</li>
<li>Not Supported: 21742/56297 (38.6%)</li>
</ul></details>


</ul></details>

JaxLinAMD commented on src/core/settings_core.json:
```diff
@@ -2019,4 +2036,4 @@
       "Description": "Maximum string length for a miscellaneous string setting"
     }
   ]
-}
```
Hi @samikhawaja could you tell me how you made this change in your editor? I can't tell the difference between the two lines.

JakeMerdichAMD replied:
That's just a trailing newline. No functional change.

samikhawaja replied:
The coding standard above doesn't say anything about the trailing new line. I can remove it if this is not ok.

JaxLinAMD replied:
this is ok, no need to change it any more :)

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -138,6 +139,24 @@ InternalMemMgr::InternalMemMgr(
 // Explicitly frees all GPU memory allocations.
 void InternalMemMgr::FreeAllocations()
 {
+
```
```suggestion
```

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,129 @@ uint32 InternalMemMgr::GetReferencesCount()
+// =====================================================================================================================
+// Map the GPU memory allocation for cpu access
+Result InternalMemMgr::Map(
+        GpuMemory* pGpuMemory,
```
```suggestion
    GpuMemory* pGpuMemory,
```

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,129 @@ uint32 InternalMemMgr::GetReferencesCount()
+// Map the GPU memory allocation for cpu access
+Result InternalMemMgr::Map(
+        GpuMemory* pGpuMemory,
+        void** ppData)
```
```suggestion
    void** ppData)
```

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,129 @@ uint32 InternalMemMgr::GetReferencesCount()
+// =====================================================================================================================
+// Unmap the GPU memory allocation from cpu address space
+Result InternalMemMgr::Unmap(
+        GpuMemory* pGpuMemory)
```
```suggestion
    GpuMemory* pGpuMemory)
```

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,129 @@ uint32 InternalMemMgr::GetReferencesCount()
+{
+    if (m_pDevice->Settings().maxMappedPoolsSize >= 0)
+    {
+        while ((m_totalSizeMappedPools > m_pDevice->Settings().maxMappedPoolsSize)
```
```suggestion
        while ((m_totalSizeMappedPools > m_pDevice->Settings().maxMappedPoolsSize) &&
```

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +582,129 @@ uint32 InternalMemMgr::GetReferencesCount()
+    if (m_pDevice->Settings().maxMappedPoolsSize >= 0)
+    {
+        while ((m_totalSizeMappedPools > m_pDevice->Settings().maxMappedPoolsSize)
+                && (m_unusedMappedPoolList.NumElements() > 0))
```
```suggestion
                (m_unusedMappedPoolList.NumElements() > 0))
```

JaxLinAMD commented on src/core/internalMemMgr.h:
```diff
@@ -56,6 +56,8 @@ struct GpuMemoryPool
     uint64                          pagingFenceVal;         // Paging fence value
 
     Util::BuddyAllocator<Platform>* pBuddyAllocator;        // Buddy allocator used for the suballocation
+    void*                           pData;                  // address of the already existing mapping
```
```suggestion
    void*                           pData;                  // Address of the already existing mapping
```

JaxLinAMD commented on src/core/internalMemMgr.h:
```diff
@@ -56,6 +56,8 @@ struct GpuMemoryPool
 
     Util::BuddyAllocator<Platform>* pBuddyAllocator;        // Buddy allocator used for the suballocation
+    void*                           pData;                  // address of the already existing mapping
+    size_t                          refCount;               // refCount the number of memory allocations use this mapping
```
```suggestion
    size_t                          refCount;               // Tracks the number of memory allocations use this mapping
```

JaxLinAMD commented on src/core/internalMemMgr.h:
```diff
@@ -115,6 +118,17 @@ class InternalMemMgr
 
+    Result Map(
+        GpuMemory* pGpuMemory,
+        void** ppData);
```
```suggestion
        void**     ppData);
```

JaxLinAMD commented on src/core/internalMemMgr.h:
```diff
@@ -115,6 +118,17 @@ class InternalMemMgr
+    // If the number of mapped pools are more then the maximum limit then unmap the least recently used pool.
+    void CheckMappedPoolLimit();
+
+
```
```suggestion
```

JaxLinAMD commented on src/core/internalMemMgr.cpp:
```diff
@@ -563,4 +581,129 @@ uint32 InternalMemMgr::GetReferencesCount()
+// Map the GPU memory allocation for cpu access
+Result InternalMemMgr::Map(
+    GpuMemory* pGpuMemory,
+    void** ppData)
```
```suggestion
    void**     ppData)
```

JaxLinAMD commented:
retest this please

JaxLinAMD commented:
retest this please

JaxLinAMD commented:
retest this please

JaxLinAMD commented:
/rebase

JaxLinAMD commented:
@samikhawaja could you rebase your branch？

samikhawaja commented:
> @samikhawaja could you rebase your branch？

Sure. Rebased the branch on dev.

amdvlk-admin commented:
### Test summary for commit e129c531cc0b6b08b170d67b68075fd3c629030c
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev 665ea8970f411bb9cbb08be8e75016d8cb09a4d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c6950e03a4c97beadc88a8bf5bea3dc5aa2649ee</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32381/56344 (57.5%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23962/56344 (42.5%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

amdvlk-admin commented:
e129c531cc0b6b08b170d67b68075fd3c629030c Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/pal-github-pr/driver_build/drivers/pal/src/core/libInit.cpp:64:95: error: static assertion failed: The specified PAL_CLIENT_INTERFACE_MAJOR_VERSION is not supported.
   64 |     static_assert((PAL_CLIENT_INTERFACE_MAJOR_VERSION >= PAL_MINIMUM_INTERFACE_MAJOR_VERSION) &&
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
   65 |                   (PAL_CLIENT_INTERFACE_MAJOR_VERSION <= PAL_INTERFACE_MAJOR_VERSION),
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~          
[222/3024] Building CXX object pal/CMakeFiles/pal.dir/src/core/internalMemMgr.cpp.o

amdvlk-admin commented:
e129c531cc0b6b08b170d67b68075fd3c629030c Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/pal-github-pr/driver_build/drivers/pal/src/core/libInit.cpp:64:95: error: static assertion failed: The specified PAL_CLIENT_INTERFACE_MAJOR_VERSION is not supported.
   64 |     static_assert((PAL_CLIENT_INTERFACE_MAJOR_VERSION >= PAL_MINIMUM_INTERFACE_MAJOR_VERSION) &&
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
   65 |                   (PAL_CLIENT_INTERFACE_MAJOR_VERSION <= PAL_INTERFACE_MAJOR_VERSION),
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~          
[221/3091] Building CXX object pal/CMakeFiles/pal.dir/src/core/image.cpp.o
